src/models/match.model.ts

```diff
@@ -5,10 +5,15 @@ export const MatchSchema = z.object({
     homeTeam: z.string(),
     awayTeam: z.string(),
     date: z.date(),
-    competition: z.string(),
+    league: z.string(),
     country: z.string(),
     stadium: z.string(),
     createdAt: z.date(),
```

Comment:
```suggestion
    export const zodDate = z.string().transform((str) => new Date(str));
    createdAt: zodDate,
```
It knows to take string from json of date and convert it to date object so you wont have to anytime you use it. z.date() dont parses response successfully of get all reacommendations